engine/crates/engine-v2/schema/tests/conversion.rs

```diff
@@ -1,101 +1,101 @@
-
-directive @join__graph(name: String!, url: String!) on ENUM_VALUE
-
-directive @join__implements(graph: join__Graph!, interface: String!) repeatable on OBJECT | INTERFACE
-
-directive @join__type(graph: join__Graph!, key: join__FieldSet, extension: Boolean! = false, resolvable: Boolean! = true, isInterfaceObject: Boolean! = false) repeatable on OBJECT | INTERFACE | UNION | ENUM | INPUT_OBJECT | SCALAR
-
-directive @join__unionMember(graph: join__Graph!, member: String!) repeatable on UNION
-
-directive @link(url: String, as: String, for: link__Purpose, import: [link__Import]) repeatable on SCHEMA
-
-scalar join__FieldSet
-
-enum join__Graph {
-  ACCOUNTS @join__graph(name: "accounts", url: "http://accounts:4001/graphql")
-  INVENTORY @join__graph(name: "inventory", url: "http://inventory:4002/graphql")
-  PRODUCTS @join__graph(name: "products", url: "http://products:4003/graphql")
-  REVIEWS @join__graph(name: "reviews", url: "http://reviews:4004/graphql")
-}
-
-scalar link__Import
-
-enum link__Purpose {
-  """
-  `SECURITY` features provide metadata necessary to securely resolve fields.
-  """
-  SECURITY
-
-  """
-  `EXECUTION` features provide metadata necessary for operation execution.
-  """
-  EXECUTION
-}
-
-type Product
-  @join__type(graph: INVENTORY, key: "upc")
-  @join__type(graph: PRODUCTS, key: "upc")
-  @join__type(graph: REVIEWS, key: "upc")
-{
-  upc: String!
-  weight: Int @join__field(graph: INVENTORY, external: true) @join__field(graph: PRODUCTS)
-  price: Int @join__field(graph: INVENTORY, external: true) @join__field(graph: PRODUCTS)
-  inStock: Boolean @join__field(graph: INVENTORY)
-  shippingEstimate: Int @join__field(graph: INVENTORY, requires: "price weight")
-  name: String @join__field(graph: PRODUCTS)
-  reviews: [Review] @join__field(graph: REVIEWS)
-}
-
-type Query
-  @join__type(graph: ACCOUNTS)
-  @join__type(graph: INVENTORY)
-  @join__type(graph: PRODUCTS)
-  @join__type(graph: REVIEWS)
-{
-  me: User @join__field(graph: ACCOUNTS)
-  user(id: ID!): User @join__field(graph: ACCOUNTS)
-  users: [User] @join__field(graph: ACCOUNTS)
-  topProducts(first: Int = 5): [Product] @join__field(graph: PRODUCTS)
-}
-
-type Review
-  @join__type(graph: REVIEWS, key: "id")
-{
-  id: ID!
-  body: String
-  product: Product
-  author: User @join__field(graph: REVIEWS, provides: "username")
-}
-
-type User
-  @join__type(graph: ACCOUNTS, key: "id")
-  @join__type(graph: REVIEWS, key: "id")
-{
-  id: ID!
-  name: String @join__field(graph: ACCOUNTS)
-  username: String @join__field(graph: ACCOUNTS) @join__field(graph: REVIEWS, external: true)
-  birthday: Int @join__field(graph: ACCOUNTS)
-  reviews: [Review] @join__field(graph: REVIEWS)
-}
-"#;
-
-#[test]
-fn should_not_fail() {
-    let FederatedGraph::V1(graph) = FederatedGraph::from_sdl(SCHEMA).unwrap();
-    let config = config::VersionedConfig::V1(graph).into_latest();
-    let _schema = Schema::from(config);
-}
+// use engine_v2_schema::Schema;
```

Comment:
This test is broken, because `from_sdl()` and `to_sdl()`  now work with FederatedGraph::V2, so we need the engine v2 conversion code to unbreak it.

Reply:
IMHO you should just an `engine_v2_config::Config::from_graph(federated_graph::latest::FederatedGraph)`

Reply:
I left the conversion code alone (so far) in this PR, so it still expects a federated graph v1 — we have to change that, probably even before merging this branch or tests are going to be broken.

Reply:
ahh right, I can push a commit in your branch for that if you want once you're done, up to you